ocaml/idl/datamodel.ml

```diff
@@ -5249,14 +5251,47 @@ let network_detach_for_vm = call
     ~allowed_roles:_R_VM_POWER_ADMIN
     ()
 
+let network_purpose = Enum ("network_purpose", [
+  "nbd", "Network Block Device service using TLS";
+  "insecure_nbd", "Network Block Device service without integrity or confidentiality: NOT RECOMMENDED";
+  "management", "Management";
+  "storage", "Storage";
+  "guest", "Guest virtual machines";
+  "himn", "Host internal management network on which guests will be assigned a private link-local IP address which can be used to talk XenAPI";
+  "unmanaged", "Not managed by xapi";
```

Comment:
The network class has a managed field, so maybe it's best not to duplicate that here, for now: https://xapi-project.github.io/xen-api/classes/network.html
Maybe in the future we could deprecate that field, and just make the getter & setter inspect / change the purposes field of the network.

Reply:
Fair point.